java/Makefile

```diff
@@ -1,25 +1,29 @@
-ifeq ($(JAVA_HOME),)
-  $(warning No JAVA_HOME found, JNI building will fail.  Please set JAVA_HOME when using JNI)
-  exit 1
-endif
-
 VWLIBS := -L../vowpalwabbit -l vw
 STDLIBS = $(BOOST_LIBRARY) $(LIBS)
 JAVA_INCLUDE = -I $(JAVA_HOME)/include
 
 ifeq ($(UNAME), Linux)
   JAVA_INCLUDE += -I $(JAVA_HOME)/include/linux
+  SHARED_LIBRARY = libvw_jni.so
```

Comment:
you might want to abstract out the library name `vw_jni` as well, since `System.loadLibrary("vw_jni");` will compose the shared library name based on it.